### ISSUE TITLE
Mortal Integration?

### ISSUE THREAD

**Issue by xingfanxia** [Jimboom7/AlphaJong#106]
Pure strategy is pretty weak, maybe can integrate with a local mortal instance?[Equim-chan/Mortal](https://github.com/Equim-chan/Mortal)

**Comment by wizardt** [comment]
Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.

https://github.com/747929791/MajsoulAI
https://github.com/747929791/majsoul_wrapper

**Comment by wizardt** [comment]
> Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> 
> https://github.com/747929791/MajsoulAI https://github.com/747929791/majsoul_wrapper

Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.

https://github.com/747929791/MajsoulAI
https://github.com/747929791/majsoul_wrapper

**Comment by xingfanxia** [comment]
> > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> 
> Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> 
> [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)

Ah ok

**Comment by xingfanxia** [comment]
> > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> 
> Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> 
> [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)

What about majsoul-helper? This seems also strategy based but is much more capable. https://github.com/EndlessCheng/mahjong-helper

I was able to get this to work for majsoul with this mid layer https://github.com/Avenshy/mahjong-helper-majsoul-mitmproxy. Maybe can make it fully automated by fetching strategy from the helper

**Comment by wizardt** [comment]
> > > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> > 
> > 
> > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> 
> What about majsoul-helper? This seems also strategy based but is much more capable. https://github.com/EndlessCheng/mahjong-helper
> 
> I was able to get this to work for majsoul with this mid layer https://github.com/Avenshy/mahjong-helper-majsoul-mitmproxy. Maybe can make it fully automated by fetching strategy from the helper

Yes, mahjong-helper, it works for me too. But if we want a AI(machine learning) based automated bot play for us, there isn't one(for free or open source). There are AI bots that some people are selling them for money which can do the job.

**Comment by thtTNT** [comment]
I'm not sure if Mortal exposes their models. If they do, I think the idea is technically feasible. However, I don't think it's a good idea to open source this modified project, as it might facilitate cheating.

**Comment by DivineSolar** [comment]
> 

I was actually looking for one although I can't seem to find a paid bot for sale. Perhaps its just because it may be in another language.

**Comment by DivineSolar** [comment]
> > > > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > > > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> > > 
> > > 
> > > Mortal is not open-sourse. There is wrapper and MajsoulAI, but all seems outdated and no one are developing.
> > > [747929791/MajsoulAI](https://github.com/747929791/MajsoulAI) [747929791/majsoul_wrapper](https://github.com/747929791/majsoul_wrapper)
> > 
> > 
> > What about majsoul-helper? This seems also strategy based but is much more capable. https://github.com/EndlessCheng/mahjong-helper
> > I was able to get this to work for majsoul with this mid layer https://github.com/Avenshy/mahjong-helper-majsoul-mitmproxy. Maybe can make it fully automated by fetching strategy from the helper
> 
> Yes, mahjong-helper, it works for me too. But if we want a AI(machine learning) based automated bot play for us, there isn't one(for free or open source). There are AI bots that some people are selling them for money which can do the job.

I was actually looking for one although I can't seem to find a paid bot for sale. Perhaps its just because it may be in another language. But as for the English search results I usually only get about 4 - 8 pages total with no paid bots that I could find utilizing the machine tech.
